### PR TITLE
Port transitions.dart to fn3

### PR DESCRIPTION
**Pull request by abarth**
@Hixie

### CODE REVIEW

**Comment by abarth** [comment]
I added some more widgets to this PR.

File: sky/packages/sky/lib/src/fn3/dismissable.dart [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+}
+
+typedef void ResizedCallback();
+typedef void DismissedCallback();
+
+class Dismissable extends StatefulComponent {
+  Dismissable({
+    Key key,
+    this.child,
+    this.onResized,
```
Comment (Hixie) [inline sky/packages/sky/lib/src/fn3/dismissable.dart]: rename all the onResized stuff to be "want resize" or something along those lines. It's asking to be resized, not actually resizing itself, at least in the case of being in a MixedViewport.

File: sky/packages/sky/lib/src/fn3/focus.dart [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    this.scopeFocused: true, // are we focused in our ancestor scope?
+    this.focusedScope, // which of our descendant scopes is focused, if any?
+    this.focusedWidget,
+    Widget child
+  }) : super(key: key, child: child);
+
+  final bool scopeFocused;
+  final FocusState focusState;
+
+  // These are mutable because we implicitly changed them when they're null in
```
Comment (Hixie) [inline sky/packages/sky/lib/src/fn3/focus.dart]: changed -> change

File: sky/packages/sky/lib/src/fn3/framework.dart [hunk 3]
```diff
@@ -335,6 +335,11 @@ abstract class ComponentState<T extends StatefulComponent> {
   /// additional state when the config field's value is changed.
   void didUpdateConfig(T oldConfig) { }
 
+  /// Called when this object is inserted into the tree. Override this function
+  /// to perform initialization that depends on the location at which this
+  /// object was inserted into the tree.
+  void didMount(BuildContext context) { }
```
Comment (Hixie) [inline sky/packages/sky/lib/src/fn3/framework.dart]: call this initState for better symmetry with dispose().

**Comment by Hixie** [comment]
LGTM

**Pull request merged**